client/components/Editor/views/CodeView.ts

```diff
@@ -112,6 +124,15 @@ export default class CodeBlockView {
 		if (node.type !== this.node.type) return false;
 		this.node = node;
 		if (this.updating) return true;
+		const newLanguageName = node.attrs.language;
+		if (newLanguageName !== this.currentLanguageName) {
+			languageModes[newLanguageName].importMode().then((languageMode) => {
+				this.cm.dispatch({
+					effects: activeLanguageMode.reconfigure(languageMode[newLanguageName]()),
+				});
```

Comment:
Maybe we could extract this into a function like `loadAndDispatchLanguageMode` that imports the language module and reconfigures the language mode compartment.